pkg/appgw/requestroutingrules.go

```diff
@@ -273,12 +273,20 @@ func (c *appGwConfigBuilder) getSslRedirectConfigResourceReference(targetListene
 	return resourceRef(sslRedirectConfigID)
 }
 
-func (c *appGwConfigBuilder) modifyPathRulesForRedirection(httpURLPathMap *n.ApplicationGatewayURLPathMap, targetListener listenerIdentifier) {
-	// Application Gateway supports Basic and Path-based rules
+func (c *appGwConfigBuilder) modifyPathRulesForRedirection(cbCtx *ConfigBuilderContext, httpURLPathMap *n.ApplicationGatewayURLPathMap, targetListener listenerIdentifier) {
+
+	// We could end up in a situation where we are attempting to attach a redirect, which does not exist.
+	redirectRef := c.getSslRedirectConfigResourceReference(targetListener)
+	redirectsSet := *c.getRedirectsByID(c.getRedirectConfigurations(cbCtx))
```

Comment:
We should be able to use `c.appGw.RedirectConfiguration` at this point ?

Reply:
We agreed that we should have pure functions and not have to rely on certain order of operations for this to be correct.